src/org/skypro/skyshop/product/SearchEngine.java

```diff
@@ -1,40 +1,34 @@
 package org.skypro.skyshop.product;
 
 import java.util.Arrays;
+import java.util.LinkedList;
 
 public class SearchEngine {
 
-    private final Searchable[] searchables;
+    LinkedList<Searchable> searchables;
```

Comment:
поля должны быть приватные всегда. Также тип ссылки должен быть максимально общим (см. полиморфизм)
```suggestion
    private List<Searchable> searchables;
```